git_machete/client.py

```diff
@@ -1682,20 +1682,21 @@ def checkout_github_prs(self, pr_no: int) -> None:
         remote: str
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
         debug('checkout_github_pr()', f'organization is {org}, repository is {repo}')
+        print(f"Fetching {remote}...")
+        self.__git.fetch_remote(remote)
+        if self.__git.remotes():
+            self.flush_caches()
 
-        pr = get_pull_request_by_number(str(pr_no), org, repo)
+        pr = get_pull_request_by_number_or_none(str(pr_no), org, repo)
         if not pr:
             raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`")
-
+        if '/'.join([remote, pr.head]) not in self.__git.get_remote_branches():
+            raise MacheteException(f"Could not checkout PR #{pr_no} because it's head branch `{pr.head}` is already deleted from `{remote}`.")
```

Comment:
```suggestion
            raise MacheteException(f"Could not check out PR #{pr_no} because its head branch `{pr.head}` is already deleted from `{remote}`.")
```